frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -929,6 +1048,15 @@ pub mod pallet {
 	#[pallet::getter(fn snapshot_metadata)]
 	pub type SnapshotMetadata<T: Config> = StorageValue<_, SolutionOrSnapshotSize>;
 
+	/// Sorted (worse -> best) list of unchecked, signed solutions.
+	#[pallet::storage]
+	#[pallet::getter(fn signed_submissions)]
+	pub type SignedSubmissions<T: Config> = StorageValue<
+		_,
+		Vec<SignedSubmission<T::AccountId, BalanceOf<T>, CompactOf<T>>>,
+		ValueQuery,
+	>;
```

Comment:
Each signed submission could be very large no?

Do we access all of them at once at any time? if not, you should probably make some kind of queue using a map instead.

Reply:
The two fundamental operations we do with signed submissions are [insert](https://github.com/paritytech/substrate/blob/fd383561da9b0fe655f986cd86e231583c78e6a2/frame/election-provider-multi-phase/src/signed.rs#L170) and [finalize](https://github.com/paritytech/substrate/blob/fd383561da9b0fe655f986cd86e231583c78e6a2/frame/election-provider-multi-phase/src/signed.rs#L63-L64). The latter case involves accessing them all, though it's true that we don't actually need all members simultaneously. I'll defer to your storage experience, but my impression was that it's more efficient to store a `StorageValue<Vec<_>>` than a `StorageMap` when we do need to iterate over all members.

I've got https://github.com/paritytech/substrate/pull/8933 in progress to replace the `Vec` with a `BoundedBTreeSet`. I guess that in principle we could emulate some kind of linked list with a `StorageMap` instead, but my impression is that doing so just wouldn't be efficient.

Reply:
fwiw I think since the queue of signed submissions will be fairly small (maximum of 5 or 10 solutions), `vec` is just fine.